engine/crates/engine-v2/id-derives/src/lib.rs

```diff
@@ -0,0 +1,12 @@
+mod id;
+mod indexes;
+
+#[proc_macro_derive(Id, attributes(max))]
+pub fn derive_id(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
+    id::derive_id(input)
+}
+
+#[proc_macro_derive(IndexImpls, attributes(indexed_by))]
```

Comment:
maybe `IndexedFields`?

Reply:
Tbh the more I dug into how your `proxy` stuff worked (which this was intended to replace) the less I thought it was a good idea...  Far too magic

So not sure this is either